### ISSUE TITLE
Feature: PI: export PI setting in a convenient format to R

### ISSUE THREAD

**Issue by StephanSchaller** [Open-Systems-Pharmacology/PK-Sim#1582]
For customized PI, it would be convenient, if the PI setup could be exported in a convenient format that could then be used externally within R

**Comment by Yuri05** [comment]
In the previously available Matlab PI Export the whole PI configuration was exported as XML.

Additionally, a minimal matlab code was exported, which could be modified by the user then.
The Matlab toolbox included the functions for evaluation of the error function (saved within the XML PI configuration) and so on.

Example: [PI_MatlabExportExample.zip](https://github.com/Open-Systems-Pharmacology/PK-Sim/files/5455153/PI_MatlabExportExample.zip)

Generated Matlab code:
```
% Name of file that contains the definition of the identification problem
PI_xml = 'Parameter Identification 1.xml';

% List of corresponding simulations
simulationList = {'Henrist oral Xanthium as weibull.xml', 'Kaumeier IV 100 mg.xml'};

% read xml and initialize simulations with corresponding parameters
PI=initParameterIdentificationForOSPSuiteExport(PI_xml,simulationList);
 
% start the algorithm
finalValues = fminsearch(@(p) getPIErrorFunctionalForOSPSuiteExport(p,PI),[PI.par(:).startValue]);

% create default output figures for the parameter identification
plotPIDefaultOutputForOSPSuiteExport(PI,finalValues);
```

So what would be the expectation for the R PI Export? Same principle?

**Comment by StephanSchaller** [comment]
> So what would be the expectation for the R PI Export? Same principle?

Not sure, yet. We are currently looking into this from the R side (maybe with a bridge to Python). Once we know more we will formulate some requirements (but I guess the former Matlab-based approach is a good basis).

@abdelr FYI

**Comment by PavelBal** [comment]
>So what would be the expectation for the R PI Export? Same principle?

Configuration in some format - either as XML, or R-code. If XML, ospsuiteR should implement methods to read such an XML and generate R - data structures from it (e.g., R6-classes?). Additionally, simulations used for the PI in PKML format.

The configuration should at least include:
 - All the data sets used for PI (at this step, we could also implement reading observed data PKMLs in ospsuiteR and export data as pkml)
- Mapping of model outputs to observed data.
- Parameters to estimate (paths?), including their start values and boundaries.

**Comment by Yuri05** [comment]
Some thoughts on this:
1. I would prefer exporting configuration independent from the programming language. E.g XML/JSON/... but not R
2. Thinking about OSP models compiled as native C++ code: there should be an easy possibility in the R-Code used for the PI to switch between pkml models and compiled models
(Currently only _Matlab_-interface to compiled models exists, but I hope to create an R-interface in the mid-term future)
3. PI implementation in R should provide the possibility to save the PI-result (parameter values / molecules initial values) in a format that can be easily imported by PK-Sim/MoBi

**Comment by PavelBal** [comment]
> I would prefer exporting configuration independently from the programming language. E.g XML/JSON/... but not R
Agree. I think JSON is good, as it is already used e.g. for snapshots.

> PI implementation in R should provide the possibility to save the PI-result (parameter values / molecules initial values) in a format that can be easily imported by PK-Sim/MoBi

For MoBi, prarameter start values and MSV can be imported from an excel-file. For PK-Sim, I am not aware of such a  functionality.

**Comment by PavelBal** [comment]
PI is part of JSON snapshot. This is all we need.
